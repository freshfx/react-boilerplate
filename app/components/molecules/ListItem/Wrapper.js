import styled from '@emotion/styled'

const Wrapper = styled.li`
  width: 100%;
  height: 3em;
  display: flex;
  align-items: center;
  position: relative;
  border-top: 1px solid #eee;

  &:first-of-type {
    border-top: none;
  }
`

export default Wrapper
