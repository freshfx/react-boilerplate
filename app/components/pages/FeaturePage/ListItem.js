import styled from '@emotion/styled'

const ListItem = styled.li`
  margin: 1em 0;
`

export default ListItem
