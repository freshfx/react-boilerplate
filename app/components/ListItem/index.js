import React from 'react'
import PropTypes from 'prop-types'

import Item from './Item'
import Wrapper from './Wrapper'

const ListItem = ({item}) => (
  <Wrapper>
    <Item>{item}</Item>
  </Wrapper>
)

ListItem.propTypes = {
  item: PropTypes.any
}

ListItem.defaultProps = {
  item: null
}

export default ListItem
