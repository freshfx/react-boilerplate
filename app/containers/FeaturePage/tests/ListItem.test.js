import 'jest-styled-components'
import React from 'react'
import {mount} from 'enzyme'

import ListItem from '../ListItem'

const renderComponent = (props = {}) => mount(<ListItem {...props} />)

describe('<ListItem />', () => {
  it('should match snapshot', () => {
    const renderedComponent = renderComponent()
    expect(renderedComponent).toMatchSnapshot()
  })
})

