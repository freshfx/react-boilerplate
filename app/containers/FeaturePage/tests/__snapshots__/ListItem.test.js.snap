// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should match snapshot 1`] = `
.c0 {
  margin: 1em 0;
}

<ListItem>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "ListItem-sc-27v6xc-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "margin:1em 0;",
          ],
        },
        "displayName": "ListItem",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "ListItem-sc-27v6xc-0",
        "target": "li",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <li
      className="c0"
    />
  </StyledComponent>
</ListItem>
`;
